### ISSUE TITLE
[feature] midway 是否支持客户端 http 接口，然后再调用 grpc 。类似 BFF 

### ISSUE THREAD

**Issue by ishowman** [midwayjs/midway#977]
如题，看 grpc 章节是在 midway 去调用 grpc 的方法。想问后续是否会有相关方案，实现 midway 提供 serveless 能力给客户端，处理客户端请求后再去调用 grpc 呢？有的话需要怎么做？
或者你们有更好的方案推荐？欢迎探讨。

* **Node Version**:
* **Egg Version**:
* **Plugin Name**:
* **Plugin Version**:
* **Platform**:
* **Mini Showcase Repository**:

**Comment by czy88840616** [comment]
没看懂。

**Comment by ishowman** [comment]
> 没看懂。grpc 支持提供 grpc 服务和调用 grpc服务两种。

前端应用和 midway 不在同个参考，分开部署的话要怎么调用？不能像传统的 api 应该通过 http + json 去请求接口吧？

**Comment by czy88840616** [comment]
明白了，你需要http网关对接grpc服务？

**Comment by ishowman** [comment]
> 明白了，你需要http网关对接grpc服务？

对。我的想法是要么搭建 BFF 层去提供，或者将 ts 版本的 rpc 整成 npm package，使用 npm package 里的方法去调用接口(这个感觉不好实现)。
我开发经验不多，不知道哪一个方案更好，希望能听取你的建议。
如果 midway 后续能提供开箱即用的方案作为最佳实践，那就很方便了。 thanks。

**Comment by riskgod** [comment]
> > 明白了，你需要http网关对接grpc服务？
> 
> 对。我的想法是要么搭建 BFF 层去提供，或者将 ts 版本的 rpc 整成 npm package，使用 npm package 里的方法去调用接口(这个感觉不好实现)。
> 我开发经验不多，不知道哪一个方案更好，希望能听取你的建议。
> 如果 midway 后续能提供开箱即用的方案作为最佳实践，那就很方便了。 thanks。

https://github.com/midwayjs/midway/issues/977#issuecomment-817236722 这个解释我觉得可以，对于架构的话 我目前的实现方案是把鉴权 熔断 放在网关层做，BFF层做 http和grpc的转化，后端的服务全部用grpc写。用node写grpc 链路追踪这块有点麻烦，我是自己用requestID来追踪，可以参考下这套架构。

**Comment by xmsz** [comment]
> 我目前的实现方案是把鉴权 熔断 服务发现，注册等 放在网关层做

想知道你的网关层是用什么实现的
